spynnaker/pyNN/models/neural_projections/connectors/abstract_connector.py

```diff
@@ -67,7 +67,7 @@ def set_space(self, space):
         """
         self._space = space
 
-    def _set_weights_and_delays(self, weights, delays, allow_lists):
+    def set_weights_and_delays(self, weights, delays):
```

Comment:
I actually think this function is necessary - I think the confusion has arisen because the merge with master didn't work properly, so every other function had delays / weights as an argument, when in fact in master they don't use them any more.  But it's necessary for a different reason now - it's the function that's being used to pass in the weights and delays from the synapse information in each case.  (And in the case of the FromListConnector, making sure that the weights and/or delays supplied by the user in their list don't get overwritten).

Reply:
I have looked back at this in previous iterations.  It doesn't look like connectors in the master version ever took in the weight or delay parameters, so I think these have been added here...

Reply:
Now I’m confused - I’m sure at least some of the functions here did take weights and/or delays as parameters - (_get_delay_maximum(..) and _get_weight_maximum(..) certainly did...), but looking back it does look as though I added a number of them on this branch.  I’m of the opinion that using the set_weights_and_delays(..) function here to send the weights and delays in is a better way of doing it than passing them into the functions, but if you have an opinion as to which is best then we can go with that...

Reply:
(And of course overriding this function in the FromListConnector as required...)

Reply:
The underscore functions did and do, as these are static utility functions in the abstract class.  The other functions didn't but probably should - this is what is needed to avoid storing the weights and delays in the classes, which is needed to allow the connector to be used in multiple projections.  This *could* be done by requiring that each call to the functions is preceded by a call to set_weights_and_delays but this seems error prone, as it will only take one missed call to make it all fall over.